### DISCUSSION TITLE
Is there away to retrieve the number of photons/electrons from an image?

### DISCUSSION THREAD

**Discussion by Odotocodot** [mitsuba-renderer/mitsuba3#1050] [Q&A]
Hello

I was wondering if it possible to retrieve the number of photons from an image.
I have been following this discussion https://github.com/mitsuba-renderer/mitsuba3/discussions/411#discussion-4611022, and have been using `Film.develop(raw=True)`. However, im not sure what the values represent in the returned `TensorXf` for all channels.

I am using a `specfilm` with a `thinlens` sensor. My `srf` is only one band, with two wavelengths (500 to 501nm)

Any help is much appreciated!

**Comment by njroussel** [comment]
Hi @Odotocodot 

You should have a 2-channel Tensor. The first channel is the weighted radiance of the band you're measuring, and the second is the weight channel. Although represented by a floating point value, in practice (with default settings) it should be the total number of "valid" ray paths for that position.

**Reply by Odotocodot** [reply]
Thanks for the reply.

I see, then can each valid ray then be thought of as a photon then? If so would the number of photons be calculated by multiplying by the previous channel? This value seems very small.

**Reply by njroussel** [reply]
> I see, then can each valid ray then be thought of as a photon then?

Yes

> If so would the number of photons be calculated by multiplying by the previous channel? This value seems very small.

I obviously don't know what you're rendering but in most cases it will be constant and equivalent to the `spp` value you pass to `mi.render()` or that is set in your scene description.
